observability/test/o11y_tests_manager.py

```diff
@@ -0,0 +1,306 @@
+                )
+
+    def generate_gke_resource_identifier(self) -> str:
+        return '%s-%s' % (self.args.job_mode,
+                          ''.join(random.choices(string.ascii_lowercase + string.digits, k=6)))
+
+    def add_job_to_job_queue(self,
+                             queue_key: QueueKey,
+                             server_lang: SupportedLang,
+                             client_lang: SupportedLang,
+                             test_case: ObservabilityTestCase) -> None:
+        self.job_queues[queue_key].add_job(TestJob(
+            server_lang = server_lang,
+            client_lang = client_lang,
+            test_case = test_case,
+            job_name = '%s:%s:%s' % (server_lang, client_lang, test_case),
+            port = self.curr_port_num,
+        ))
+        self.curr_port_num += 1
+
+    def get_next_interop_shard_num(self, entry_key: InteropEntryKey) -> int:
+        num = self.curr_interop_shard_nums[entry_key]
+        self.curr_interop_shard_nums[entry_key] = (num + 1) % NUM_INTEROP_QUEUES
+        return num
+
+    def get_next_lang_shard_num(self, lang: SupportedLang) -> int:
+        num = self.curr_lang_shard_nums[lang]
+        self.curr_lang_shard_nums[lang] = (num + 1) % NUM_LANG_QUEUES
+        return num
+
+    def add_interop_jobs(self) -> None:
+        for entry in INTEROP_COMBINATIONS:
+            entry_key = InteropEntryKey('%s:%s' % (entry['server_lang'], entry['client_lang']))
+            for test_case in INTEROP_TEST_CASES:
+                queue_key = QueueKey('%s:%d' % (entry_key, self.get_next_interop_shard_num(entry_key)))
+                self.add_job_to_job_queue(queue_key,
+                                          SupportedLang(entry['server_lang']),
+                                          SupportedLang(entry['client_lang']),
+                                          test_case)
+
+    def add_lang_jobs(self, lang: SupportedLang) -> None:
+        for test_case in ObservabilityTestCase:
+            queue_key = QueueKey('%s:%d' % (lang, self.get_next_lang_shard_num(lang)))
+            self.add_job_to_job_queue(queue_key, lang, lang, test_case)
+
+    def start_process_in_background(self,
+                                    job: TestJob,
+                                    queue_key: QueueKey,
+                                    extra_args: List[str] = []) -> subprocess.Popen:
+        sponge_log_dir = TestUtil.get_sponge_log_dir(self.args.job_mode, job.job_name)
+        # Note: the corresponding sponge_log.xml is being written in run_o11y_tests.py itself
+        with open(os.path.join(sponge_log_dir, 'sponge_log.log'), 'a') as out:
+            logger.info('Starting %s, in queue %s' % (job.job_name, queue_key))
+            env = os.environ.copy()
+            env['GRPC_VERBOSITY'] = 'DEBUG'
+            return subprocess.Popen(
+                [os.path.join(os.path.dirname(__file__), 'run_o11y_tests.py'),
```

Comment:
Nit: Instead of relying on this file being executable an defining a shebang, it might be better to use [`sys.executable`](https://docs.python.org/3/library/sys.html#sys.executable). This might avoid potential mismatches between desired Python binary used and the actual binary used.

Reply:
Implemented.